kctf-operator/pkg/controller/challenge/challenge_controller.go

```diff
@@ -146,154 +113,67 @@ func (r *ReconcileChallenge) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
+	// Set default values not configured by kubebuilder
+	SetDefaultValues(challenge)
+
 	// Check if the deployment already exists, if not create a new one
 	found := &appsv1.Deployment{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: challenge.Name,
+	err = r.client.Get(ctx, types.NamespacedName{Name: challenge.Name,
 		Namespace: challenge.Namespace}, found)
 
-	// Set default values not configured by kubebuilder
-	SetDefaultValues(challenge)
-
 	// Just enters here if it's a new deployment
 	if err != nil && errors.IsNotFound(err) {
 		// Define a new deployment
-		dep := r.deploymentForChallenge(challenge)
-		reqLogger.Info("Creating a new Deployment", "Deployment.Namespace",
-			dep.Namespace, "Deployment.Name", dep.Name)
-		err = r.client.Create(context.TODO(), dep)
-
-		if err != nil {
-			reqLogger.Error(err, "Failed to create new Deployment", "Deployment.Namespace",
-				dep.Namespace, "Deployment.Name", dep.Name)
-			return reconcile.Result{}, err
-		}
-
-		// Deployment created successfully - return and requeue
-		return reconcile.Result{Requeue: true}, nil
+		return r.CreateDeployment(challenge, ctx)
```

Comment:
I like that you moved these to functions. Makes it much more readable IMO :)

Reply:
:)